C_CPP/Algorithmic Approaches/Backtracking/m-coloring/mColoring.cpp

```diff
@@ -0,0 +1,85 @@
+#include<iostream>
+#include<fstream>
+using namespace std;
+
+int n;// for storing number of vertices
+int m;// maximum numbers of colours that can be used
+int x[100];//contains numbers from 1 to m .x[k] represent the colour assigned to kth vertex
+int graph[100][100];//storing adjancy matrix
+int cnt=0;// for counting number of answers possible
+
+void nextval(int k)
+{
+	while(true)
+	{
+		x[k]=(x[k]+1)%(m+1);//next higher colour
+		if(x[k]==0) return;//all colours exhausted
+		int j;
+		for(j=1 ; j<=k-1 ; ++j){
+			if(graph[k][j]==1 and x[j]==x[k]) //colour adjacent to it have same colour
+                break;

```

Comment:
Make it properly indented, like this
                if(graph[k][j]==1 and x[j]==x[k]) //colour adjacent to it have same colour
                break;